gradio/chat_interface.py

```diff
@@ -267,14 +268,12 @@ def _setup_events(self) -> None:
         submit_triggers = [self.textbox.submit]
 
         submit_event = (
-            on(
-                submit_triggers,
+            self.textbox.submit(
                 self._clear_and_save_textbox,
-                [self.textbox],
-                [self.textbox, self.saved_input],
+                [self.textbox, self.previous_input],
+                [self.textbox, self.saved_input, self.previous_input],
                 show_api=False,
                 queue=False,
-                preprocess=False,
```

Comment:
Used `preprocess=False` in the past so that the data would be of `MultimodalData` instance. But for some reason, state variables do not get passed to events that have `preprocess=False`. Something to look into but I just added back preprocessing.